### PR TITLE
Implement GetStatus without synchronization

### PR DESCRIPTION
**Pull request by dinhl**
Task: https://github.com/hwsc-org/hwsc-file-transaction-svc/issues/30

This pull request implements `GetStatus` without synchronization.
The changes cover the following:
- Fix display errors
- Add global variable for class State
- Add documentation
- Adhere to PEP8
- Change CRLF

### CODE REVIEW

File: service.py [hunk 1]
```diff
@@ -15,34 +20,44 @@ class Servicer(hwsc_file_transaction_svc_pb2_grpc.FileTransactionServiceServicer
...
+                 print("[INFO] Requesting GetStatus service")
 
-             #TODO
-             def DownloadZippedFiles(self, request_iterator, context):
-                 """Download zipped files from azrue blob storage."""
-                 if request_iterator.name:
-                     return utility.download_chunk(self.tmp_file_name)
+                 if(grpc.StatusCode.OK.value[0]!= State.AVAILABLE.value):
+                     context.set_code = grpc.StatusCode.UNAVAILABLE.value[0]
+                     context.set_details = grpc.StatusCode.UNAVAILABLE.value[1]
```
Comment (faraonc) [inline service.py]: to access the name you use `grpc.StatusCode.OK.name`

File: service.py [hunk 2]
```diff
@@ -15,34 +20,44 @@ class Servicer(hwsc_file_transaction_svc_pb2_grpc.FileTransactionServiceServicer
...
-                 """Get and return status to the user."""
-                 print("Get Status")
+                 print("[INFO] Requesting GetStatus service")
 
-             #TODO
-             def DownloadZippedFiles(self, request_iterator, context):
-                 """Download zipped files from azrue blob storage."""
-                 if request_iterator.name:
-                     return utility.download_chunk(self.tmp_file_name)
+                 if(grpc.StatusCode.OK.value[0]!= State.AVAILABLE.value):
```
Comment (faraonc) [inline service.py]: unnecessary parenthesis

Comment (faraonc) [inline service.py]: You dont have a global variable that holds the state of the service. You are just comparing two constants. Please check this https://github.com/hwsc-org/hwsc-document-svc/blob/master/service/service.go#L25

File: service.py [hunk 3]
```diff
@@ -15,34 +20,44 @@ class Servicer(hwsc_file_transaction_svc_pb2_grpc.FileTransactionServiceServicer
...
+                     context.set_code = grpc.StatusCode.UNAVAILABLE.value[0]
+                     context.set_details = grpc.StatusCode.UNAVAILABLE.value[1]
+
+                 else:
+                     context.set_code = grpc.StatusCode.OK.value[0]
+                     context.set_details = grpc.StatusCode.OK.value[1]
+
+                 return hwsc_file_transaction_svc_pb2.FileTransactionResponse(
+                     code=context.set_code,
+                     message=context.set_details
```
Comment (faraonc) [inline service.py]: This file by default uses `CRLF`. Please convert to `LF`.

File: service.py [hunk 4]
```diff
@@ -5,6 +5,11 @@
 import io
 import hwsc_file_transaction_svc_pb2
 import hwsc_file_transaction_svc_pb2_grpc
+from enum import Enum
+
+class State(Enum):
```
[the remark this answers is not in the document]
Comment (dinhl) [reply service.py]: """current state of the service"""
class State(Enum):
    AVAILABLE = 0
    UNAVAILABLE = 1

File: service.py [hunk 5]
```diff
@@ -1,107 +1,142 @@
...
 import grpc
-import time
 import utility
-import io
 import hwsc_file_transaction_svc_pb2
 import hwsc_file_transaction_svc_pb2_grpc
+from enum import Enum
+
+
+state = 0
```
Comment (faraonc) [inline service.py]: the correct way is
```
class State(Enum):
    """A current state class of file transaction service"""
    AVAILABLE = 0
    UNAVAILABLE = 1


state = State.AVAIALBLE

```

File: service.py [hunk 6]
```diff
@@ -1,107 +1,142 @@
...
+            def __init__(self):
+                pass
+
+            def GetStatus(self, request, context):
+                """Return the status of the service"""
+                print("[INFO] Requesting GetStatus service")
+                global state
+                state = State.AVAILABLE.value
+                available = grpc.StatusCode.OK.value[0]
+
```
Comment (faraonc) [inline service.py]: `GetStatus` should not mutate the `state`. It only checks the `state`. 

`global state` this line is good enough

line 33 and line 34 are NOT needed

Comment (dinhl) [reply service.py]: if i delete line 33, 34, the if statement will look like this, does it ok ?

if grpc.StatusCode.OK.value[0] != state.value

Comment (faraonc) [reply service.py]: ` if state != State.AVAILABLE:`

File: service.py [hunk 7]
```diff
@@ -1,107 +1,137 @@
...
+        self.server.start()
+        print("[INFO] hwsc-file-transaction-svc initializing...")
+        print("[INFO] hwsc-file-transaction started at:")
+        print(port)
+
+        try:
+            while True:
+                pass
+        except KeyboardInterrupt:
+            self.server.stop(0)
```
Comment (faraonc) [inline service.py]: you need a blank new line after this line

File: service.py [hunk 8]
```diff
@@ -1,107 +1,137 @@
...
+
+class State(Enum):
+    """A current state class of file transaction service"""
+    AVAILABLE = 0
+    UNAVAILABLE = 1
+
+
+class FileTransactionService(hwsc_file_transaction_svc_pb2_grpc.FileTransactionServiceServicer):
+    """A FileTransactionService class contains servicer and corresponding functionalites."""
+
```
Comment (faraonc) [inline service.py]: add `state = State.AVAILABLE` in here

File: service.py [hunk 9]
```diff
@@ -1,107 +1,137 @@
...
+    def __init__(self):
+
+        class Servicer(hwsc_file_transaction_svc_pb2_grpc.FileTransactionServiceServicer):
+            """A servicer class contains functionalities for file service."""
+
+            def __init__(self):
+                self.state = State.AVAILABLE
```
Comment (faraonc) [inline service.py]: remove `self.state = State.AVAILABLE`

File: service.py [hunk 10]
```diff
@@ -1,107 +1,137 @@
...
+                pass
+
+            def GetStatus(self, request, context):
+                """Return the status of the service"""
+                print("[INFO] Requesting GetStatus service")
+
+                if self.state != State.AVAILABLE:
```
Comment (faraonc) [inline service.py]: `if FileTransactionService.state != State.AVAILABLE`

File: test_utility.py [hunk 11]
```diff
@@ -1,22 +1,47 @@
...
+    assert utility.verify_uuid("1234abcd5454efef8842ll3fsc") == True
+
+    # testing with shorter length uuid
+    assert utility.verify_uuid("1234abcd5454efef8842ll3fs") == False
+
+    # testing with longer length uuid
+    assert utility.verify_uuid("1234abcd5454efef8842llaaaab") == False
+
+    # testing with special case in uuid
+    assert utility.verify_uuid("1234abcd5454efef8842ll3!ss") == False
```
Comment (faraonc) [inline test_utility.py]: add new line

File: test_utility.py [hunk 12]
```diff
@@ -1,22 +1,47 @@
...
-def test_valid_uuid2():
-    with pytest.raises(Exception) as e_info:
-        utility.create_uuid_container_in_azure("1234abcd5454efef8842ll3!ss")
-
-# should be not passed. but how?
-def test_valid_uuid4():
-    with pytest.raises(Exception) as e_info:
-        utility.create_uuid_container_in_azure("1234abcd5454efef8842llaaaa")
\ No newline at end of file
+def test_get_file_type():
```
Comment (faraonc) [inline test_utility.py]: a lot of pep8 errors

**Pull request merged**